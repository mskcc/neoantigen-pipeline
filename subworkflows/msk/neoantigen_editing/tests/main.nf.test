nextflow_workflow {

    name "Test Workflow NEOANTIGEN_EDITING"
    script "../main.nf"
    workflow "NEOANTIGEN_EDITING"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/neoantigen_editing"
    tag "neoantigeninput"
    tag "neoantigenediting/aligntoiedb"
    tag "neoantigenediting/computefitness"


    test("neoantigen - test") {

        when {
            workflow {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['patient_data'], checkIfExists: true)                   
                    ]
                input[1] = file(params.test_data_mskcc['neoantigen']['iedb']['iedb_fasta'], checkIfExists: true)

                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.versions,
                    file(workflow.out.annotated_output[0][1]).name
                    ).match()
                }           
            )

        }
    }
}
