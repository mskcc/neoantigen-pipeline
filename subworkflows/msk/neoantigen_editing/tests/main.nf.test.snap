{
    "neoantigen - test": {
        "content": [
            [
                "versions.yml:md5,37984f930810748dc81d61b73680e10a",
                "versions.yml:md5,563eced4337e36c41c7450059792e907"
            ],
            "3_OLTS_primary_tumor_annotated.json"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-22T14:44:00.937168"
    }
}